controllers/cloud.redhat.com/namespacepool_controller.go

```
@@ -91,9 +91,8 @@ func (r *NamespacePoolReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		}
 	}
 
-	if err := r.client.Status().Update(ctx, &pool); err != nil {
-		r.log.Error(err, fmt.Sprintf("cannot update [%s] pool status", pool.Name))
-		return ctrl.Result{}, err
+	if err = r.client.Status().Update(ctx, &pool); err != nil {
+		return ctrl.Result{Requeue: true}, nil
```

I think you also want to return the error, no?